tests/periph_uart/tests/periph_uart_if.py

```diff
@@ -22,11 +22,16 @@ def uart_init(self, dev=DEFAULT_DEV, baud=DEFAULT_BAUD):
         self.send_cmd("send {} {}".format(dev, "flush"))
         return ret
 
-    def uart_mode(self, data_bits, parity, stop_bits, dev=DEFAULT_DEV):
+    def uart_mode(self, data_bits, parity, stop_bits, dev=DEFAULT_DEV,
+                  baud=DEFAULT_BAUD):
         """Setup databits, parity and stopbits."""
-        return self.send_cmd(
+        self.send_cmd("init {} {}".format(dev, baud))
```

Comment:
we should only wrap API calls and not change (or add) functionality

Reply:
Agreed.

Reply:
Hmm many I can change the command to `uart_init_mode`?  Again this should happen every time.  The thing is many boards allow uart mode to be changed and not needing to reinit.  I think it is just the STM that has this constraint... (the registers are only or'ed not cleared with an and then or'ed)

Reply:
sounds like this should be fixed in RIOT in the cpu specific uart_mode implementation?

Reply:
okay as a quick fix lets have a separate call `uart_init_mode` in the python interface that does that, i.e. calling `mode` and `init` afterwards. but do not change (or extend) the existing `init` and `mode` wrappers, okay?